File: wellpass-backend/src/main/java/com/varol/WellPass_Mananagement_System/service/organization/ServiceProviderService.java
package com.varol.WellPass_Mananagement_System.service.organization;

import java.util.List;

import com.varol.WellPass_Mananagement_System.model.organization.ServiceProvider;

public interface ServiceProviderService {
    
    ServiceProvider createServiceProvider(ServiceProvider serviceProvider);
    
    ServiceProvider updateServiceProvider(Long providerId, ServiceProvider serviceProvider);
    
    ServiceProvider getServiceProviderById(Long providerId);
    
    ServiceProvider getServiceProviderByCode(String providerCode);
    
    List<ServiceProvider> getAllActiveServiceProviders();
    
    List<ServiceProvider> getServiceProvidersByLocation(String location);
    
    void deactivateServiceProvider(Long providerId);
    
    void activateServiceProvider(Long providerId);
    
    boolean existsByProviderCode(String providerCode);
}

File: wellpass-frontend/services/serviceProviderService.ts
import api from '@/lib/api';
import { ServiceProvider, CreateServiceProviderRequest, UpdateServiceProviderRequest } from '@/types/serviceProvider';
import { ApiResponse, PageResponse } from '@/types/api';

export const serviceProviderService = {
  async getAllServiceProviders(page = 0, size = 10) {
    const response = await api.get<ApiResponse<PageResponse<ServiceProvider>>>(
      `/api/service-providers?page=${page}&size=${size}`
    );
    return response.data.data;
  },

  async getActiveServiceProviders() {
    const response = await api.get<ApiResponse<ServiceProvider[]>>('/api/service-providers/active');
    return response.data.data;
  },

  async getServiceProviderById(id: number) {
    const response = await api.get<ApiResponse<ServiceProvider>>(`/api/service-providers/${id}`);
    return response.data.data;
  },

  async getServiceProviderByCode(code: string) {
    const response = await api.get<ApiResponse<ServiceProvider>>(`/api/service-providers/code/${code}`);
    return response.data.data;
  },

  async createServiceProvider(data: CreateServiceProviderRequest) {
    const response = await api.post<ApiResponse<ServiceProvider>>('/api/service-providers', data);
    return response.data.data;
  },

  async updateServiceProvider(id: number, data: UpdateServiceProviderRequest) {
    const response = await api.put<ApiResponse<ServiceProvider>>(`/api/service-providers/${id}`, data);
    return response.data.data;
  },

  async deleteServiceProvider(id: number) {
    const response = await api.delete<ApiResponse<void>>(`/api/service-providers/${id}`);
    return response.data;
  },

  async activateServiceProvider(id: number) {
    const response = await api.put<ApiResponse<void>>(`/api/service-providers/${id}/activate`);
    return response.data;
  },

  async deactivateServiceProvider(id: number) {
    const response = await api.put<ApiResponse<void>>(`/api/service-providers/${id}/deactivate`);
    return response.data;
  },
};